featuretools/entityset/entity.py

```diff
@@ -117,9 +117,8 @@ def __eq__(self, other, deep=False):
             return False
         if len(self.variables) != len(other.variables):
             return False
-        for v in self.variables:
-            if v not in other.variables:
-                return False
+        if set(self.variables) != set(other.variables):
```

Comment:
I removed most of the places where this would be hit, but I thought I might as well make this comparison faster while we're looking at it.